docs/TAP_SUBMISSION_REQUIREMENTS.md

```diff
@@ -0,0 +1,160 @@
+            "metadata": metadata.get_standard_metadata(
+                schema=schema,
+                key_properties=stream.key_properties,
+                valid_replication_keys=stream.replication_keys,
+                replication_method=stream.replication_method,
+            ),
+            "key_properties": stream.key_properties
+        }
+```
+(see [here][adroll-discovery] for the full example)
+
+## Stream Selection
+It's required that the tap implements stream selection so that in the Stitch product the user has control over what endpoints are hit.
+
+In order to implement stream selection, the tap must Iterate over only selected streams.
+
+The tap can get a list of the selected stream objects by calling `get_selected_streams()` on a singer-python
+`Catalog` object, as is done here:
+
+```
+selected_streams = catalog.get_selected_streams(state)
+
+for stream in selected_streams:
+    <sync stream>
+ ```
+(see [here][adroll-sync] for an example)
+
+## Field Selection
+It's required that the tap implements field selection so that in the Stitch product the user has control over which fields in the record are returned.
+
+To filter a record's fields using the selected metadata from the catalog, the supported approach is to pass every
+record through the transformer with a metadata dictionary, as is done here:
+
+```
+with Transformer() as transformer:
+    for rec in stream_object.sync():
+        singer.write_record(
+            <tap stream id>,
+            transformer.transform(
+                <record>, <schema dict>, <metadata dict>,
+            )
+        )
+```
+
+(see [here][adroll-transformer] for an example)
+
+
+## How to handle child streams
+If there are child streams, this means that they rely on some piece of information from a corresponding parent. To
+handle this, the tap must grab parent ids in child stream sync function.
+
+(see [here][adroll-streams] in the ClientStream sync method)
+
+
+## Pylint
+
+In order to submit the tap, it's required that pylint is passing at 100% using the following disables:
+`broad-except,chained-comparison,empty-docstring,fixme,invalid-name,line-too-long,missing-class-docstring,missing-function-docstring,missing-module-docstring,no-else-raise,no-else-return,too-few-public-methods,too-many-arguments`
+
+The following command should be run from the root of the repo:
+
+```
+pylint tap_<tap_name> --disable 'broad-except,chained-comparison,empty-docstring,fixme,invalid-name,line-too-long,missing-class-docstring,missing-function-docstring,missing-module-docstring,no-else-raise,no-else-return,too-few-public-methods,too-many-arguments'
+```
+
+If other disables are necessarys, use inline disables:
+
+``` python
+def get_format_values(self): # pylint: disable=no-self-use
+    return []
+```
+See [Source][trello-streams] for example.
+
+
+## Pypi
+
+In order to run in Stitch, we require sole ownership a Pypi repo using the specific naming convention: `tap-<tap-name>`
```

Comment:
> In order to run in Stitch, we require sole ownership a Pypi repo

Should this be "In order to control the releases of Singer projects, we require sole ownership of the PyPI repo"?

Because:
1. We used to try and not mention Stitch in the Singer docs, is that still our convention?
2. It does feel like less of a Stitch issue and more of a "what makes it into the codebase" issue